lib/Cake/Network/CakeSocket.php

```diff
@@ -129,8 +129,12 @@ public function connect() {
 			$this->disconnect();
 		}
 
+		$hasProtocol = strpos($this->config['host'], '://') !== false;
```

Comment:
I used string slicing because I didn't think of parse_url() when I wrote the code. It looks like parse_url() would be more complicated though.